backend/src/core/server.ts

```diff
@@ -50,26 +40,21 @@ export default class Server {
 
   private setupSockets(): void {
     const io = socketIo(this.httpServer);
-    io.on(Event.CONNECT, socket => {
+    io.on(Event.CONNECT, (socket: ExtendedSocket) => {
+      socket.username = "";
       logger.debug(`Socket ${socket.id} connected.`);
 
       socket.on(Event.JOIN_ROOM, roomId => {
         new RoomSocketHandler(this.database, io, socket, roomId).initialize();
       });
 
-      socket.on(Event.DISCONNECT, socket => {
+      socket.on(Event.DISCONNECT, (socket: ExtendedSocket) => {
         logger.debug(`Socket ${socket.id} disconnected.`);
       });
 
-      socket.on(Event.CREATE_USERNAME, username => {
-        const extSocket = socket as ExtendedSocket;
-        if (username === "") {
-          const randomName: string = uniqueNamesGenerator(customNameConfig);
-          extSocket.username = `Anonymous ${randomName}`;
-        } else {
-          extSocket.username = username;
-        }
-        logger.info(`socket username set to ${extSocket.username}`);
+      socket.on(Event.CREATE_USERNAME, (username: string) => {
```

Comment:
I think this should probably be in `handler.ts`

Reply:
Moved! :D thank you